src/wp-includes/block-editor.php

```diff
@@ -0,0 +1,280 @@
+			'icon'  => null,
+		),
+		array(
+			'slug'  => 'embed',
+			'title' => _x( 'Embeds', 'block category' ),
+			'icon'  => null,
+		),
+		array(
+			'slug'  => 'reusable',
+			'title' => _x( 'Reusable Blocks', 'block category' ),
+			'icon'  => null,
+		),
+	);
+
+	/**
+	 * Filters the default array of categories for block types.
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param array[] $default_categories Array of categories for block types.
+	 */
+	$block_categories = apply_filters( "block_categories_{$editor_name}", $default_categories );
+	if ( 'post-editor' === $editor_name ) {
+		$post = is_object( $editor_name_or_post ) ? $editor_name_or_post : get_post();
+
+		/**
+		 * Filters the default array of categories for block types.
+		 *
+		 * @since 5.0.0
+		 * @deprecated 5.8.0 The hook transitioned to support also screens that don't contain the $post instance.
+		 *
+		 * @param array[] $block_categories Array of categories for block types.
+		 * @param WP_Post $post             Post being loaded.
+		 */
+		$block_categories = apply_filters_deprecated( 'block_categories', array( $block_categories, $post ), '5.8.0', "block_categories_{$editor_name}" );
+	}
+
+	return $block_categories;
+}
+
+/**
+ * Gets the list of allowed block types to use in the block editor.
+ *
+ * @since 5.8.0
+ *
+ * @param string $editor_name The name of the editor (e.g. 'post-editor').
+ *
+ * @return bool|array Array of block type slugs, or boolean to enable/disable all.
+ */
+function get_allowed_block_types( $editor_name ) {
+	$allowed_block_types = true;
+
+	/**
+	 * Filters the allowed block types for the given editor, defaulting to true (all
+	 * registered block types supported).
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param bool|array $allowed_block_types Array of block type slugs, or
+	 *                                        boolean to enable/disable all.
+	 */
+	$allowed_block_types = apply_filters( "allowed_block_types_{$editor_name}", $allowed_block_types );
+	if ( 'post-editor' === $editor_name ) {
+		$post = get_post();
+
+		/**
+		 * Filters the allowed block types for the editor, defaulting to true (all
+		 * block types supported).
+		 *
+		 * @since 5.0.0
+		 * @deprecated 5.8.0 The hook transitioned to support also screens that don't contain $post instance.
+		 *
+		 * @param bool|array $allowed_block_types Array of block type slugs, or
+		 *                                        boolean to enable/disable all.
+		 * @param WP_Post    $post                The post resource data.
+		 */
+		$allowed_block_types = apply_filters_deprecated( 'allowed_block_types', array( $allowed_block_types, $post ), '5.8.0', "allowed_block_types_{$editor_name}" );
+	}
+
+	return $allowed_block_types;
+}
+
+/**
+ * Gets the default block editor settings.
+ *
+ * @since 5.8.0
+ *
+ * @param string $editor_name The name of the editor (e.g. 'post-editor').
+ *
+ * @return array The default block editor settings.
+ */
+function get_default_block_editor_settings( $editor_name ) {
```

Comment:
It feels like `get_default_block_editor_settings` shouldn't be concerned with the name of the editor, we could offer a higher level function like `get_block_editor_settings` that would call `get_default_block_editor_settings` internally and set `allowedBlockTypes` and `blockCategories` that might differ depending on the editor type.

In addition to that the new `get_block_editor_settings` method could internally process Global Styles and call the `block_editor_settings` filter. @nosolosw and @youknowriad, what do you think?

Reply:
> It feels like get_default_block_editor_settings shouldn't be concerned with the name of the editor, we could offer a higher level function like get_block_editor_settings that would call get_default_block_editor_settings internally and set allowedBlockTypes and blockCategories that might differ depending on the editor type.

I think it's important to have some default values for these two properties but it's ok if the filter is run after this function and overrides the default values.

> In addition to that the new get_block_editor_settings method could internally process Global Styles and call the block_editor_settings filter. @nosolosw and @youknowriad, what do you think?

I'm ok with that, for the post editor, we'll have to run the  filter twice (once with editor_name= "post-editor" and once without  any editor_name or  using something like "pre-global-styles" or similar)

Reply:
> > It feels like get_default_block_editor_settings shouldn't be concerned with the name of the editor, we could offer a higher level function like get_block_editor_settings that would call get_default_block_editor_settings internally and set allowedBlockTypes and blockCategories that might differ depending on the editor type.
> I think it's important to have some default values for these two properties but it's ok if the filter is run after this function and overrides the default values.

Right, there is an existing `get_block_categories` and the new `get_allowed_block_types` that use filters internally so we can keep them inside `get_default_block_editor_settings` but they expect `$editor_name` to be flexible. It's maybe less useful with block categories, but for the list of the allowed blocks, it's essential.

Reply:
I'm trying an approach where `get_default_block_editor_settings ` is a lowe level function and the editor type agnostic. It provides defaults for the block categories and allowed block types plus all other shared settings. `get_block_editor_settings` expects the editor name to be provided, overrides `allowedBlockTypes` and `blockCategories` if filters are applied. In addition to that, it runs filters on all block editor settings.